x-pack/plugins/observability_ai_assistant/server/functions/context.ts

```diff
@@ -12,42 +12,23 @@ import * as t from 'io-ts';
 import { last, omit } from 'lodash';
 import { lastValueFrom } from 'rxjs';
 import { FunctionRegistrationParameters } from '.';
-import { MessageRole, type Message } from '../../common/types';
+import { FunctionVisibility, MessageRole, type Message } from '../../common/types';
 import { concatenateChatCompletionChunks } from '../../common/utils/concatenate_chat_completion_chunks';
 import type { ObservabilityAIAssistantClient } from '../service/client';
 
-export function registerRecallFunction({
+export function registerContextFunction({
   client,
   registerFunction,
   resources,
 }: FunctionRegistrationParameters) {
   registerFunction(
     {
-      name: 'recall',
+      name: 'context',
       contexts: ['core'],
-      description: `Use this function to recall earlier learnings. Anything you will summarize can be retrieved again later via this function.
-      
-      The learnings are sorted by score, descending.
-      
-      Make sure the query covers ONLY the following aspects:
-      - Anything you've inferred from the user's request, but is not mentioned in the user's request
-      - The functions you think might be suitable for answering the user's request. If there are multiple functions that seem suitable, create multiple queries. Use the function name in the query.  
-
-      DO NOT include the user's request. It will be added internally.
-      
-      The user asks: "can you visualise the average request duration for opbeans-go over the last 7 days?"
-      You recall: {
-        "queries": [
-          "APM service,
-          "lens function usage",
-          "get_apm_timeseries function usage"    
-        ],
-        "categories": [
-          "lens",
-          "apm"
-        ]
-      }`,
-      descriptionForUser: 'This function allows the assistant to recall previous learnings.',
+      description: '',
+      visibility: FunctionVisibility.Internal,
+      descriptionForUser:
+        'This function allows the assistant to recall previous learnings from the Knowledge base and gather context of how you are using the application.',
```

Comment:
I don't have all the historical context (no pun intended) so just thinking out loud: Are we perhaps overloading the recall/context function with too much responsibility? Instead of changing the recall function, what about letting the consumer specify context specific functions dynamically to make them available for the LLM to call? 

The functions can be pre-registered like normally but will not be passed to the LLM unless they are specified as part of the chat context. Then the functions and important params can be declared like:

```
const { setChatContext } = useObservabilityAIAssistantChatContext();

setChatContext({
  params: {
    service: {
      value: "backend",
      description: "The service the user has selected",
    },
    environment: {
      value: "production",
      description: "The environment the user has selected",
    },
    transactionName: {
      value: "POST /cart",
      description: "The name of the selected transaction",
    },
    traceId: {
      value: "2cb494454edc437679309dc190d692368ae7da490f45aff291fb6f7989437d4a",
      description: "The id of the selected trace",
    },
  },
  functions: [
    "get_trace",
    "get_logs_for_trace",
    "get_errors_for_trace",
    "get_errors_for_transaction",
  ],
});
```